workshop/content/20-typescript/30-hello-cdk/300-cdk-watch.md

```diff
@@ -0,0 +1,213 @@
+
+1. Open the [AWS Lambda
+   Console](https://console.aws.amazon.com/lambda/home#/functions) (make sure
+   you are in the correct region).
+
+    You should see our function:
+
+    INSERT PHOTO HERE
+
+2. Click on the function name to go to the console.
+
+3. The code should be loaded onto the screen. Did your change show up?
+
+    INSERT PHOTO HERE
+
+## CDK Watch
+
+We can do better than calling `cdk deploy` or `cdk deploy --hotswap` each time.
+`cdk watch` is similar to `cdk deploy` except that instead of being a one-shot
+operation, it monitors your code and assets for changes and attempts to perform a 
+deployment automatically when a change is detected. By default, `cdk watch` will 
+use the `--hotswap` flag, which inspects the changes and determines if those 
+changes can be hotswapped. To disable this behavior, you can call
+`cdk watch --no-hotswap`.
+
+Once we set it up, we can use `cdk watch` to detect both hotswappable changes and
+changes that require full CloudFormation deployment.
+
+## Modify your `cdk.json` file
+
+When the `cdk watch` command runs, the files that it observes is determined by the
+`"watch"` setting in the `cdk.json` file. It has two sub-keys, `"include"` and
+`"exclude"`, each of which can be either a single string or an array of strings.
+Each entry is interpreted as a path relative to the location of the `cdk.json`
+file. Globs, both `*` and `**`, are allowed to be used.
+
+Your `cdk.json` file should look like this:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/cdk-workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "**/*.js",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    // ...
+  }
+}
+```
+
+As you can see, the sample app comes with a suggested `"watch"` setting. We do in
+fact want to observe our `.js` files in the `lambda` folder, so let's remove
+`"**/*.js"` from the `"exclude"` list:
+
+```json
+{
+  "app": "npx ts-node --prefer-ts-exts bin/cdk-workshop.ts",
+  "watch": {
+    "include": [
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "**/*.d.ts",
+      "tsconfig.json",
+      "package*.json",
+      "yarn.lock",
+      "node_modules",
+      "test"
+    ]
+  },
+  "context": {
+    // ...
+  }
+}
+```
+
+Now you're all set to start watching!
+
+## See how long 'CDK Watch' takes
+
+First, call `cdk watch`: 
+
+```bash
+cdk watch
+```
+
+This will trigger an initial deployment and immediately begin observing the files
+you've specified in `cdk.json`.
+
+Let's change our lambda asset code in `lambda/hello.js` one more time. It doesn't
+matter what you change it to, but let's say we want it to say
+`"Good Afternoon, CDK"` now.
+
+Once you click save or hit `CTRL+s`, `cdk watch` will recognize that your file has
```

Comment:
Let's just say "Once you save the changes to your Lambda code file", or something.